pkg/kube/resources.go

```diff
@@ -168,222 +137,77 @@ func CreateResourceProviderFromAPI(kube kubernetes.Interface, clusterName string
-		oldControllers = append(oldControllers, oldController)
-	}
-	controllersV1B2, err := kube.AppsV1beta2().StatefulSets("").List(listOpts)
-	if err != nil {
-		logrus.Errorf("Error fetching StatefulSets v1beta2: %v", err)
-		return nil, err
-	}
-	for _, oldController := range controllersV1B2.Items {
-		oldControllers = append(oldControllers, oldController)
-	}
-
-	for _, oldController := range oldControllers {
-		str, err := json.Marshal(oldController)
+	} else {
+		yamlNode := make(map[string]interface{})
+		err = yaml.Unmarshal(contentBytes, &yamlNode)
 		if err != nil {
-			logrus.Errorf("Error marshaling old StatefulSet version: %v", err)
-			return nil, err
-		}
-		controller := appsv1.StatefulSet{}
-		err = json.Unmarshal(str, &controller)
-		if err != nil {
-			logrus.Errorf("Error unmarshaling old StatefulSet version: %v", err)
-			return nil, err
-		}
-		controllers = append(controllers, controller)
-	}
-	return controllers, nil
-}
-
-func getDaemonSets(kube kubernetes.Interface) ([]appsv1.DaemonSet, error) {
-	listOpts := metav1.ListOptions{}
-	controllerList, err := kube.AppsV1().DaemonSets("").List(listOpts)
-	if err != nil {
-		logrus.Errorf("Error fetching DaemonSets: %v", err)
-		return nil, err
-	}
-	controllers := controllerList.Items
-
-	controllersV1B2, err := kube.AppsV1beta2().DaemonSets("").List(listOpts)
-	if err != nil {
-		logrus.Errorf("Error fetching DaemonSets v1beta2: %v", err)
-		return nil, err
-	}
-
-	for _, oldController := range controllersV1B2.Items {
-		str, err := json.Marshal(oldController)
-		if err != nil {
-			logrus.Errorf("Error marshaling old DaemonSet version: %v", err)
-			return nil, err
-		}
-		controller := appsv1.DaemonSet{}
-		err = json.Unmarshal(str, &controller)
-		if err != nil {
-			logrus.Errorf("Error unmarshaling old DaemonSet version: %v", err)
-			return nil, err
-		}
-		controllers = append(controllers, controller)
-	}
-	return controllers, nil
-}
-
-func getCronJobs(kube kubernetes.Interface) ([]batchv1beta1.CronJob, error) {
-	listOpts := metav1.ListOptions{}
-	controllerList, err := kube.BatchV1beta1().CronJobs("").List(listOpts)
-	if err != nil {
-		logrus.Errorf("Error fetching CronJobs: %v", err)
-		return nil, err
-	}
-	controllers := controllerList.Items
-
-	controllersV2A1, err := kube.BatchV2alpha1().CronJobs("").List(listOpts)
-	if err != nil {
-		logrus.Errorf("Error fetching CronJobs v2alpha1: %v", err)
-		return nil, err
-	}
-
-	for _, oldController := range controllersV2A1.Items {
-		str, err := json.Marshal(oldController)
-		if err != nil {
-			logrus.Errorf("Error marshaling old CronJob version: %v", err)
-			return nil, err
+			logrus.Errorf("Invalid YAML: %s", string(contents))
+			return err
 		}
-		controller := batchv1beta1.CronJob{}
-		err = json.Unmarshal(str, &controller)
+		finalDoc := make(map[string]interface{})
+		finalDoc["metadata"] = yamlNode["metadata"]
+		finalDoc["apiVersion"] = "v1"
+		finalDoc["kind"] = "Pod"
```

Comment:
I think we're losing the `Kind` from the file here. E.g. if the user has a `Deployment` manifest in their repo, we want that to show up as a `Deployment` in the dashboard.